### ISSUE TITLE
Member still raises CS1591 in the main file even though the source generator has generated the doc comment

### ISSUE THREAD

**Issue by SunnieShine** [dotnet/roslyn#54166]
**Version Used**: 
VS 16.11.0 Preview 2

**Steps to Reproduce**:

Write a source generator that not only generates the member, but also generates the doc comment, such as:

```csharp
context.AddSource(
    $"{name}.g.cs",
    $@"#pragma warning disable 1591

using System.Runtime.CompilerServices;

#nullable enable

namespace {namespaceName}
{{
    partial {typeKind} {symbol.Name}{genericParametersList}
    {{
        /// <inheritdoc cref=""object.ToString""/>
        [MethodImpl(MethodImplOptions.AggressiveInlining)]
        public override {readonlyKeyword}partial string ToString() => ToString(null, null);

        /// <summary>
        /// Returns a string that represents the current object with the specified format string.
        /// </summary>
        /// <param name=""format"">
        /// The format. If available, the parameter can be <see langword=""null""/>.
        /// </param>
        /// <returns>The string result.</returns>
        [MethodImpl(MethodImplOptions.AggressiveInlining)]
        public {readonlyKeyword}partial string ToString(string? format) => ToString(format, null);
    }}
}}"
);
```

**Expected Behavior**:

In the main file, because of the success of the code generating, the main file doesn't raise CS1591.

**Actual Behavior**:

CS1591 still raised.

![1](https://user-images.githubusercontent.com/23616315/122412935-94a97e00-cfb8-11eb-9cf8-94621cc9ed72.png)

**Comment by Youssef1313** [comment]
Related to https://github.com/dotnet/roslyn/issues/54119 and https://github.com/dotnet/roslyn/issues/41171

**Comment by jaredpar** [comment]
Agree this is a dupe of #54103
